### PR TITLE
Adding ability to secure Sentinel

### LINKED ISSUE #305: Missing ability to secure sentinel?
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 5.5.10
- Distribution: CentOS 7.6.1810
- Module version: 4.0
- Redis: >= 5.0.1

## How to reproduce (e.g Puppet code you use)
I mean  I don't have code but currently you can't configure the sentinel instances to requirepass to improve security in multi server redis instances managed by sentinel.

## What are you seeing
No security

## What behaviour did you expect instead
security via requirepass at the sentinel level

### PR DESCRIPTION
**Pull request by markasammut**
#### Pull Request (PR) description

Sentinel instances themselves can be secured by requiring clients to authenticate via the AUTH command. Starting with Redis 6.2, the Access Control List (ACL) is available, whereas previous versions (starting with Redis 5.0.1) support password-only authentication. This PR introduces the password-only authentication available from Redis 5.0.1 onwards.

https://redis.io/topics/sentinel#sentinel-password-only-authentication

#### This Pull Request (PR) fixes the following issues

Fixes #305

### CODE REVIEW

**Comment by markasammut** [comment]
Failed test does not look to be related to the Code.

File: manifests/sentinel.pp [hunk 1]
```diff
@@ -119,6 +122,7 @@
   String[1] $master_name  = 'mymaster',
   Stdlib::Host $redis_host = '127.0.0.1',
   Stdlib::Port $redis_port = 6379,
+  Optional[String] $requirepass = undef,
```
Comment (bastelfreak) [inline manifests/sentinel.pp:125]: I think an empty string doesn't make sense here? Please update it to:
```suggestion
  Optional[String[1]] $requirepass = undef,
```

**Comment by bastelfreak** [comment]
thanks for the PR! can you pleae add a unit test for this?

**Comment by markasammut** [comment]
@bastelfreak Note that CI check for Puppet 5 - Ubuntu 20.04 is failing in all PRs because there are no Puppet 5 packages in apt.puppetlabs.com for Ubuntu 20.4 (Puppet 5 is not supported for Ubuntu 20.4, see https://puppet.com/docs/puppet/5.5/system_requirements.html#platforms-with-packages). Could this check be eliminated?

**Comment by markasammut** [comment]
Any hope this could be merged @bastelfreak ?

**Pull request merged**